fennel/featuresets/featureset.py

```diff
@@ -311,9 +318,91 @@ def __eq__(self, other: Any) -> bool:
     def __str__(self) -> str:
         return self.name
 
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, name: str) -> None:
+        self._name = name
+        if self.extractor:
+            self.extractor.output_features = [name]
+
+    @property
+    def featureset_name(self) -> str:
+        return self._featureset_name
+
+    @featureset_name.setter
+    def featureset_name(self, name: str) -> None:
+        self._featureset_name = name
+        if self.extractor:
+            self.extractor.featureset = name
+
     def fqn(self) -> str:
         return self.fqn_
 
+    # Derive an extractor for the feature using the givne params
+    # use `field` (optionally with provider and default) to derive an extractor
+    # that performs a lookup on the given field
+    # use `feature` to alias this feature to the given feature
+    def extract(
+        self,
+        field: Field = None,
+        provider: Featureset = None,
+        default=None,
+        depends_on: List[Dataset] = None,
```

Comment:
The depends_on param is redundant right? The user is explicitly already telling us the dataset it depends on via field ? Since a field belongs to a specific dataset ?

Reply:
I agree that from the user's perspective this is redundant, but from our implementation, the `Dataset.field` doesn't give us the dataset info needed to access its keys. Let's discuss in person. Also, isn't this redundancy also there for normal extractor functions?